### PR TITLE
Replace deprecated functions and update dependencies

### PR DESCRIPTION
**Pull request by shft5410**
Changes:
- Updated to ejs@3.1.9 and mime@3.0.0
- Replaced several deprecated functions

### CODE REVIEW

File: index.js
Comment (shft5410) [inline index.js:1]: Changes:
- Parse URLs using the URL constructor
- protocol.interceptBufferProtocol has been replaced with protocol.handle (see [block post](https://www.electronjs.org/blog/electron-25-0#breaking-changes) and [electron docs](https://www.electronjs.org/de/docs/latest/api/protocol#protocolhandlescheme-handler))
- Return rendered HTML data in a Response object
- Distinguish between file not found error and other errors, e.g. EJS compilation errors

File: package.json
Comment (shft5410) [inline package.json:1]: Updated to:
- ejs@3.1.9
- mime@3.0.0

Note: mime@4.0.1 is available but does not support imports using require()

**Comment by bowheart** [comment]
Hey @shft5410 thanks for this! Apologies, I saw this during Christmas break and completely forgot about it. I should spend some time updating this repo to use TypeScript and linting so code style is consistent - I see some spacing is off, but don't worry about that.

I haven't been using electron (directly, only via OpenFin) the last few years. But this looks good as far as I can tell. I'll merge this and spin up a project to test before publishing `ejs-electron@3.0.0`

**Comment by bowheart** [comment]
@shft5410 Spun up a local electron project and tested the code. Looks like it's working beautifully. `ejs-electron@3.0.0` published :tada:

**Pull request merged**